packages/replay/src/integration.ts

```diff
@@ -190,25 +201,45 @@ Sentry.init({ replaysOnErrorSampleRate: ${errorSampleRate} })`,
   /** Setup the integration. */
   private _setup(): void {
     // Client is not available in constructor, so we need to wait until setupOnce
-    this._loadReplayOptionsFromClient();
+    const finalOptions = this._loadReplayOptionsFromClient(this._initialOptions);
 
     this._replay = new ReplayContainer({
-      options: this._options,
+      options: finalOptions,
       recordingOptions: this._recordingOptions,
     });
   }
 
   /** Parse Replay-related options from SDK options */
-  private _loadReplayOptionsFromClient(): void {
+  private _loadReplayOptionsFromClient(initialOptions: InitialReplayPluginOptions): ReplayPluginOptions {
     const client = getCurrentHub().getClient();
     const opt = client && (client.getOptions() as BrowserClientReplayOptions | undefined);
 
-    if (opt && typeof opt.replaysSessionSampleRate === 'number') {
-      this._options.sessionSampleRate = opt.replaysSessionSampleRate;
+    const finalOptions = { sessionSampleRate: 0, errorSampleRate: 0, ...dropUndefinedKeys(initialOptions) };
+
+    if (!opt) {
+      return finalOptions;
```

Comment:
m: Hmm in this case we'll also want to check for the warning, I guess?

Reply:
I'd argue that in this case, something more fundamental is wrong with the SDK than anything Replay specific. We should always get defined options from the client. 

Thinking about it, the typecast above unnecessarily includes `| undefined`. I think it should be good to be removed.

Reply:
argh my jet lag isn't helping here - the client _could_ be undefined which is what we're actually checking for here 😅

So yes, we probably should emit another warning there but I'd still argue that it is more a fundamental SDK problem than just the sample rates.

Reply:
Done [7314883](https://github.com/getsentry/sentry-javascript/pull/6959/commits/73148834e8e373a95677e827ef2020e426c9724f)